src/query/service/src/pipelines/processors/transforms/hash_join/join_hash_table.rs

```diff
@@ -32,62 +35,72 @@ use common_expression::HashMethodSerializer;
 use common_expression::HashMethodSingleString;
 use common_expression::RemoteExpr;
 use common_functions::BUILTIN_FUNCTIONS;
-use common_hashtable::HashMap;
+use common_hashtable::HashJoinHashMap;
 use common_hashtable::HashtableKeyable;
-use common_hashtable::ShortStringHashMap;
-use common_hashtable::StringHashMap;
+use common_hashtable::RowPtr;
+use common_hashtable::StringHashJoinHashMap;
 use common_sql::plans::JoinType;
 use ethnum::U256;
 use parking_lot::RwLock;
 
 use super::ProbeState;
 use crate::pipelines::processors::transforms::hash_join::desc::HashJoinDesc;
-use crate::pipelines::processors::transforms::hash_join::row::RowPtr;
 use crate::pipelines::processors::transforms::hash_join::row::RowSpace;
 use crate::pipelines::processors::transforms::hash_join::util::build_schema_wrap_nullable;
 use crate::pipelines::processors::transforms::hash_join::util::probe_schema_wrap_nullable;
 use crate::sessions::QueryContext;
 use crate::sessions::TableContext;
 
-pub struct SerializerHashTable {
-    pub(crate) hash_table: StringHashMap<[u8], Vec<RowPtr>>,
+pub struct SerializerHashJoinHashTable {
+    pub(crate) hash_table: StringHashJoinHashMap,
     pub(crate) hash_method: HashMethodSerializer,
 }
 
-pub struct SingleStringHashTable {
-    pub(crate) hash_table: ShortStringHashMap<[u8], Vec<RowPtr>>,
+pub struct SingleStringHashJoinHashTable {
+    pub(crate) hash_table: StringHashJoinHashMap,
     pub(crate) hash_method: HashMethodSingleString,
 }
 
-pub struct FixedKeyHashTable<T: HashtableKeyable> {
-    pub(crate) hash_table: HashMap<T, Vec<RowPtr>>,
+pub struct FixedKeyHashJoinHashTable<T: HashtableKeyable> {
+    pub(crate) hash_table: HashJoinHashMap<T>,
     pub(crate) hash_method: HashMethodFixedKeys<T>,
 }
 
-pub enum HashTable {
-    Serializer(SerializerHashTable),
-    SingleString(SingleStringHashTable),
-    KeysU8(FixedKeyHashTable<u8>),
-    KeysU16(FixedKeyHashTable<u16>),
-    KeysU32(FixedKeyHashTable<u32>),
-    KeysU64(FixedKeyHashTable<u64>),
-    KeysU128(FixedKeyHashTable<u128>),
-    KeysU256(FixedKeyHashTable<U256>),
+pub enum HashJoinHashTable {
+    Null,
```

Comment:
Is this necessary?

Reply:
Yes, we couldn't determine the size of the hash table initially, so we first set the `hash_table` to `HashJoinHashTable::Null` during initialization.